src/main/java/com/redislabs/redisai/RedisAI.java

```diff
@@ -94,7 +97,38 @@ public boolean setTensor(String key, Object tensor, int[] dimensions){
       throw new RedisAIException(ex);
     }
   }
-  
+
+  /**
+   * TS.GET key
+   *
+   * @param key
+   * @return Tensor
+   */
+  public Tensor getTensor(String key) throws JRedisAIRunTimeException {
+    try (Jedis conn = getConnection()) {
+      List<?> reply = sendCommand(conn, Command.TENSOR_GET, SafeEncoder.encode(key), Keyword.META.getRaw(), Keyword.VALUES.getRaw() ).getObjectMultiBulkReply();
+      if(reply.isEmpty()) {
+        return null;
+      }
+      if (reply.size() <6){
+        throw new JRedisAIRunTimeException("Expected a minimum Array Reply of size 6");
+      }
+      String dtypeString = SafeEncoder.encode((byte[]) reply.get(1));
```

Comment:
@filipecosta90 shouldn't you make this code more protocol safe and use the argument name in the reply for the parsing?

Reply:
making it more future proof. will submit changes in next commit